docs/excel/custom-functions-dynamic-arrays.md

```diff
@@ -0,0 +1,58 @@
+---
+ms.date: 12/16/2019
+description: Return a dynamic array from your custom function in an Office Excel add-in.
+title: Return a dynamic array from your custom function
+localization_priority: Normal
+---
+
+# Return a dynamic array from your custom function
+
+You can return multiple results from your custom function to make it a dynamic array formula. Multiple results will spill into neighboring cells. You can return results that only spill down, to the right, or both down and to the right.
+
+To create a custom function that is a dynamic array formula, it must return a two-dimensional array of values. The following example shows how to return a dynamic array that spills down.
+
+```javascript
+/**
+ * Get text values that spill down.
+ * @customfunction
+ * @returns {string[][]} A dynamic array with multiple results.
+ */
+function spillDown() {
+  return [['first'], ['second'], ['third']];
+}
+```
+
+The following example shows how to return a dynamic array that spills right. 
+
+```javascript
+/**
+ * Get text values that spill to the right.
+ * @customfunction
+ * @returns {string[][]} A dynamic array with multiple results.
+ */
+function spillRight() {
+  return [['first', 'second', 'third']];
+}
+```
+
+The following example shows how to return a dynamic array that spills both down and right.
+
+```javascript
+/**
+ * Get text values that spill both right and down.
```

Comment:
I am curious if I run an array that plops out results accidentally on top of data already in cells (i.e. trying to spill into an area where you can't spill). Will I see a "#SPILL!" error? If so might be worth putting in troubleshooting and calling out here? Unless we are assuming that people already know how these work

Reply:
It will do the #SPILL! error, but this is expected behavior, not a troubleshooting thing. The end user has to fix it by moving cells out of the way or such. I added a link to the Office help on how this behavior works, so that has plenty of info about this.